src/sentry/api/endpoints/organization_installations.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import absolute_import
+
+import six
+
+from django.core.urlresolvers import reverse
+from django.db import IntegrityError, transaction
+
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.provider import ProviderSerializer
+from sentry.exceptions import PluginError
+from sentry.plugins import bindings
+from sentry.models import Installation, Repository
+
+
+class OrganizationInstallationEndpoint(OrganizationEndpoint):
+
+    def get(self, request, organization):
+        results = []
+        for provider_id, provider_cls in bindings.get('repository.provider').all():
+            provider = provider_cls(id=provider_id)
+
+            if not provider.has_installations:
+                continue
+
+            if provider.needs_auth(request.user, for_installation=True):
+                return Response({
+                    'error_type': 'auth',
+                    'title': provider.name,
+                    'auth_url': reverse('socialauth_associate', args=[
+                        provider.installation_auth_provider or provider.auth_provider]),
+                }, status=400)
+
+            results.append(
+                serialize(
```

Comment:
this could be slow if theres multiple providers, but we can probably overlook it for now